### ISSUE TITLE
Chrome version is updated to 83 and tests are failing in it..

### ISSUE THREAD

**Issue by Aman-RateHub** [bonigarcia/webdrivermanager#486]
**Description of the problem**: Tests are failing in the updated chrome version with an error message - org.openqa.selenium.SessionNotCreatedException: session not created: This version of ChromeDriver only supports Chrome version 81

**Browser and version**: Chrome - 83

**Operating system**: Mac OSX

**WebDriverManager version**: Latest version

**Comment by davidcarvalho-ey** [comment]
Hello. 
This is reproduced on windows too. 
Chrome version 83. 
WebdriverManager version: latest.

**Comment by davidcarvalho** [comment]
Hello.
This is reproduced on windows too.
Chrome version 83.
WebdriverManager version: latest.

**Comment by davidcarvalho** [comment]
@Aman-RateHub can you try this: 
delete the .json file in : <home>\.wdm
and then run the test.

**Comment by rekhinnvs** [comment]
Even I am facing the same issue,
chrome has updated to the version 83.0.4103.61

Looks like the latest version is not available at https://chromedriver.storage.googleapis.com/

https://github.com/bonigarcia/webdrivermanager/pull/488

**Comment by WillieLouw** [comment]
How can this be fixed in Windows, Selenium Python. Chrome version 83.0.4103.61, and tried with all the latest Chromedrivers (up to V84): selenium.common.exceptions.SessionNotCreatedException: Message: session not created: This version of ChromeDriver only supports Chrome version 80

**Comment by ckempster** [comment]
This is related to the already logged caching issues . 
Duplicate of #483 
Duplicate of #484

**Comment by bonigarcia** [comment]
Is this issue still happening? If so, the only way to trace the problem is to analyse the WebDriverManager traces (`io.github.bonigarcia.wdm.*`) at `DEBUG` level. Otherwise, it is impossible to know what is happening. I mean this kind of traces:

```
2020-03-10 17:00:38 [main] DEBUG io.github.bonigarcia.wdm.Shell.runAndWaitArray(55) -- Running command on the shell: [wmic, datafile, where, name='C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe', get, Version, /value]
2020-03-10 17:00:38 [main] DEBUG io.github.bonigarcia.wdm.Shell.runAndWaitArray(57) -- Result: Version=80.0.3987.132
2020-03-10 17:00:38 [main] DEBUG i.g.bonigarcia.wdm.WebDriverManager.getDriverVersionForBrowserFromProperties(687) -- Getting driver version for chrome80 from online versions.properties
2020-03-10 17:00:39 [main] INFO  i.g.bonigarcia.wdm.WebDriverManager.detectDriverVersionFromBrowser(617) -- Using chromedriver 80.0.3987.106 (since Google Chrome 80 is installed in your machine)
2020-03-10 17:00:39 [main] DEBUG io.github.bonigarcia.wdm.Preferences.putValueInPreferencesIfEmpty(66) -- Storing preference chrome=80 (valid until 2020-03-11 17:00:39)
2020-03-10 17:00:39 [main] DEBUG io.github.bonigarcia.wdm.Preferences.putValueInPreferencesIfEmpty(66) -- Storing preference chrome80=80.0.3987.106 (valid until 2020-03-11 17:00:39)
2020-03-10 17:00:39 [main] DEBUG i.g.bonigarcia.wdm.WebDriverManager.downloadAndExport(566) -- Driver chromedriver 80.0.3987.106 found in cache
2020-03-10 17:00:39 [main] INFO  i.g.bonigarcia.wdm.WebDriverManager.exportDriver(1115) -- Exporting webdriver.chrome.driver as C:\Users\boni\.m2\repository\webdriver\chromedriver\win32\80.0.3987.106\chromedriver.exe
```

If you are not using a logger in your project (logback, log4j, etc), see a tutorial [here](https://www.stubbornjava.com/posts/logging-in-java-with-slf4j-and-logback).

**Comment by WillieLouw** [comment]
No, same problem. Using Visual Studio Code, and Python. Google automatically upgraded the browser, and Chromedriver gave same error. Only solution currently, uninstall Google Chrome, install version working with Chromedriver, find and shut off all automatic updates for Chrome.

**Comment by markloughery** [comment]
I am getting the same issue on Ubuntu 18.04. I tried deleting the cache but it is just re-downloading the v81 driver. The workaround (for whatever reason) was to specify the architecture when initializing the driver:

`        WebDriverManager.chromedriver().arch64().setup();
`

**Comment by WillieLouw** [comment]
Found a answer (stupid mistake) for Win O/S. It was assumed that when downloading the latest Chromedriver.exe and running it, that it 'installs' but it actually just runs the instance. So copied and replaced the previous file, and it worked. Still not sure why the auto updating fell flat, but at least got it working again. Thanks guys!

**Comment by andreikondratev** [comment]
There's still something wrong with XML it fails with `io.github.bonigarcia.wdm.config.WebDriverManagerException: io.github.bonigarcia.wdm.config.WebDriverManagerException: chromedriver 83.0.4103.39 for LINUX 64 not found in https://chromedriver.storage.googleapis.com/`
`userMirror` helps

**Comment by andreikondratev** [comment]
Didn't this https://github.com/bonigarcia/webdrivermanager/commit/a52d0190b7526031c3860e18fcf1347fd48b3e68 change cause the issue?

**Comment by andruhon** [comment]
@bonigarcia is it a trace you're looking for 

```
io.github.bonigarcia.wdm.config.WebDriverManagerException: io.github.bonigarcia.wdm.config.WebDriverManagerException: chromedriver 83.0.4103.39 for LINUX 64 not found in https://chromedriver.storage.googleapis.com/
  at io.github.bonigarcia.wdm.WebDriverManager.handleException(WebDriverManager.java:725)
  at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:531)
  at io.github.bonigarcia.wdm.WebDriverManager.handleException(WebDriverManager.java:722)
  at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:531)
  at io.github.bonigarcia.wdm.WebDriverManager.setup(WebDriverManager.java:277)
  at com.unimarket.uitests.jetty.wicketpagetest.Configuration.createWebDriver(Configuration.java:172)
  at com.unimarket.uitests.jetty.wicketpagetest.Configuration.getWebDriver(Configuration.java:112)
  at com.unimarket.uitests.jetty.wicketpagetest.WebPageTestBasicContext.startWebDriver(WebPageTestBasicContext.java:110)
  at com.unimarket.uitests.jetty.wicketpagetest.WebPageTestBasicContext.start(WebPageTestBasicContext.java:101)
  at com.unimarket.uitests.jetty.wicketpagetest.WebPageTestBasicContext.beforePageTests(WebPageTestBasicContext.java:66)
  at com.unimarket.uitests.junit.JettyScope.before(JettyScope.java:21)
  at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:46)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
  at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
  at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:534)
  at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:1196)
  at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:1014)
Caused by: io.github.bonigarcia.wdm.config.WebDriverManagerException: chromedriver 83.0.4103.39 for LINUX 64 not found in https://chromedriver.storage.googleapis.com/
```
?

**Comment by andruhon** [comment]
I have a feeling that there's an issue in Saxon's XPathEvaluator

**Comment by andruhon** [comment]
https://github.com/bonigarcia/webdrivermanager/issues/486 it seems like the issue is in ` xmlns="http://doc.s3.amazonaws.com/2006-03-01"` which prevents this document from being properly processed with evalator

**Comment by andruhon** [comment]
I just did some more digging:

The Webdriver Manager is using https://mvnrepository.com/artifact/xalan/xalan/2.7.2 as XML processor and it just works with current code.

Unfortunately WebDriver manager doesn't work with Saxon-PE which we use is xPath factory. The difference is that with Saxon-PE it's necessary to consider the namespace and add the NamespaceContext, whereas the xalan doesn't seem to care about namespaces at all.

**Comment by andruhon** [comment]
I created a PR showing what would make it work with Saxon https://github.com/bonigarcia/webdrivermanager/pull/503
Sorry I don't have understanding of why it currently works without namespace.

**Comment by andruhon** [comment]
`factory.setNamespaceAware(true);` brings consistency. Now it works with both Xalan and Saxon. #503

**Comment by bonigarcia** [comment]
I am completely lost with this issue. Is this problem still happening? How can it be reproduced?

**Comment by BullerEX** [comment]
For me its still happening, Chrome Browser version:83.0.4103.116 webDriverManager  version: 4.0.0, selenium web driver version 3.141.59

**Comment by frankvanderkuur** [comment]
I'm not sure if this helps you guys at all, but we've also came across a problem in de Webdrivermanager not updating to version 83 of Chrome. In our case the problem was caused by [this](https://github.com/bonigarcia/webdrivermanager/commit/0ac04394dbe6a82c49f13f22af9c3fa916b615ef#diff-1f4964536f397e8679625954c7139bd6) the 2th of May. 

Here the targetPath is renamed to cachepath. We run a maven install from Jenkins with an environment variable

`-Dwdm.targetPath="C:\CI"`

This means that after download the new version of Chrome, it places the driver in the C:\CI directory. That's the location the TA framework uses to run the Chromedriver. after renaming the targetPath to cachePath, the problem was solved for us.

**Comment by phuoctrung-ppt** [comment]
This issue similar happens with Chrome Browser v80. I resolved this the way  moves to Driver Manager. Im try using WebDriverManager.chromedriver().version("80").setup(). This work for me.

**Comment by bonigarcia** [comment]
I have just released WebDriverManager 4.1.0, it should be fixed in this version.

**Comment by abu112629** [comment]
Tests were failing today for me inspite of using the latest version available @bonigarcia. The chrome browser available in the server is version 88.

Version used as a dependency:

<dependency>
    <groupId>io.github.bonigarcia</groupId>
    <artifactId>webdrivermanager</artifactId>
    <version>4.3.1</version>
</dependency>


Please see below error :

org.openqa.selenium.SessionNotCreatedException: session not created: This version of ChromeDriver only supports Chrome version 86
17:43:16  Current browser version is 88.0.4324.104 with binary path C:\Program Files (x86)\Google\Chrome\Application\chrome.exe
17:43:16  Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:25:53'
